translator/util/sdkutil.go

```diff
@@ -25,18 +25,12 @@ const (
 
 var DetectRegion func(mode string, credsConfig map[string]string) string = detectRegion
 var DetectCredentialsPath func() string = detectCredentialsPath
-var runInAws = os.Getenv(config.RUN_IN_AWS)
 
 func DetectAgentMode(configuredMode string) string {
 	if configuredMode != "auto" {
 		return configuredMode
 	}
-
-	if runInAws == config.RUN_IN_AWS_TRUE {
```

Comment:
Can you explain why you are removing this? From what I recall, this was a pretty impactful bug for running ECS where the agent couldn't determine that it's running on an EC2 but should have.

Reply:
Yes I believe the issue was the agent was not able to scrap `metadataendpoint` on `awsvpc network` which happens on ECS using Task Definition > v1.39 because the issue happened around when the task definition new version got release. However, after adding the endpoint`ECS_CONTAINER_METADATA_URI_V4`, it would solve the issue. For upcoming release, I will talk with @sethAmazon to make an integration test based on different task definition and will help us in recognize the changes instead of relying on the flag. However, I could keeping the flag until the `integration test` got done to avoid the impact if that make sense.

Reply:
Yeah I would keep this until a integration tests is built for this.

Reply:
On second though, should not delete this since EKS Fargate does not know how to detect `onPrem` or not so would need to keep back this flag.